src/scirpy/tl/_mutational_load.py

```diff
@@ -0,0 +1,203 @@
+                    )
+                    mutations.append(mutation)
+
+            if inplace:
+                params.set_obs(f"{chain}_IMGT_V(D)J_{frequency_string}", mutations)
+
+            else:
+                mutation_df[f"{chain}_IMGT_V(D)J_{frequency_string}"] = mutations
+        try:
+            return mutation_df
+        except NameError:
+            return None
+
+    # calculate SHM up to nucleotide 312. Referring to the IMGT unique numbering scheme this includes:
+    # fwr1, cdr1, fwr2, cdr2, fwr3, but not cdr3 and fwr4
+    if region == "IMGT_V_segment":
+        for chain in chains:
+            mutations = []
+            for row in range(len(airr_df)):
+                if (airr_df.iloc[row].loc[f"{chain}_{sequence_alignment}"] is None) or (
+                    airr_df.iloc[row].loc[f"{chain}_{germline_alignment}"] is None
+                ):
+                    mutations.append(np.nan)
+                else:
+                    v_region_germline = airr_df.iloc[row].loc[f"{chain}_{germline_alignment}"][:312]
+                    v_region_sequence = airr_df.iloc[row].loc[f"{chain}_{sequence_alignment}"][:312]
+
+                    mutation = simple_hamming_distance(
+                        v_region_sequence, v_region_germline, frequency=frequency, ignore_chars=ignore_chars
+                    )
+                    mutations.append(mutation)
+
+            if inplace:
+                params.set_obs(f"{chain}_v_segment_{frequency_string}", mutations)
+
+            else:
+                mutation_df[f"{chain}_v_segment_{frequency_string}"] = mutations
+        try:
+            return mutation_df
+        except NameError:
+            return None
+
+    if region == "subregion":
+        # subregion_df = get_airr(params, ["fwr1", "fwr2", "fwr3", "fwr4", "cdr1", "cdr2", "cdr3"], chains)
+
+        for chain in chains:
+            airr_df[f"{chain}_junction_len"] = [len(a) for a in airr_df[f"{chain}_junction"]]
+
+            mutation_dict = defaultdict(list)
+
+            for row in range(len(airr_df)):
+                if (airr_df.iloc[row].loc[f"{chain}_{sequence_alignment}"] is None) or (
+                    airr_df.iloc[row].loc[f"{chain}_{germline_alignment}"] is None
+                ):
+                    for k in list(mutation_dict.keys()):
+                        mutation_dict[k].append(np.nan)
+                else:
+                    regions = {
+                        "fwr1": (0, 78),
+                        "cdr1": (78, 114),
+                        "fwr2": (114, 165),
+                        "cdr2": (165, 195),
+                        "fwr3": (195, 312),
+                        "cdr3": (312, 312 + airr_df.iloc[row].loc[f"{chain}_junction_len"] - 6),
+                        "fwr4": (
+                            312 + airr_df.iloc[row].loc[f"{chain}_junction_len"] - 6,
+                            len(airr_df.iloc[row].loc[f"{chain}_{germline_alignment}"]),
+                        ),
```

Comment:
Do you have a reference for this calculation? 

From [Lefranc 2023](https://pubmed.ncbi.nlm.nih.gov/12477501/):

> rearranged CDR3-IMGT of 13 amino acids (and toa JUNCTION of 15 amino acids, 2nd-CYS 104 and J-TRP or J-PHE 118 being included in the JUNCTION definition). This numbering is convenient to use since 80% of the IMGT/LIGM-DBIG and TR rearranged sequences have a CDR3 IMGT length less than or equal to 13 amino acids  (IMGT statistics, October 2001). For rearranged CDR3-IMGT less than 13 amino acids, gaps are created from the top of the loop, in the following order 111, 112, 110, 113, 109, 114, etc. (Table 4A). For rearranged CDR3-IMGT more than 13 amino acids, additional positions are created, between positions 111 and 112 at the top of the CDR3-IMGT loop, in the following order 112.1, 111.1, 112.2, 111.2, 112.3, 111.3, etc. (Table 4B). 


From that text, I'd rather deduce something like

```
312 + max(junction_length, 13 * 3) # * 3 because 13 amino acids into nucleotides
```

But I haven't looked at the actual data myself.